libbpfgo/helpers/kernel_features.go

```diff
@@ -0,0 +1,111 @@
+package helpers
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+type KernelConfig map[string]string
+
+// NewKernelConfig simply creates the underlying map used by
+// the KernelConfig, you must use InitKernelConfig to read
+// in the values of the config
+func NewKernelConfig() KernelConfig {
+	kconfig := make(map[string]string)
+	return kconfig
+}
+
+// InitKernelConfig attempts to read the kernel config
+//  into the KernelConfig from either
+// - /proc/config-$(uname -r)
+// - /boot/config.gz
+func (k KernelConfig) InitKernelConfig() error {
+
+	err := k.getBootConfig()
+	if err == nil {
+		return nil
+	}
```

Comment:
I assume you wanted to check `if err != nil { return err }`?

Reply:
Nope, some kernels are compiled with `CONFIG_IKCONFIG=y` which puts the config file at /proc/config.gz instead of `/boot/config-$(uname -r)`, so the err==nil is there to fall through